game/src/contrib/components/InventoryComponent.java

```diff
@@ -11,47 +11,53 @@
 import java.util.logging.Logger;
 
 /**
- * This Component marks an {@link Entity} as having an inventory
+ * Allows the entity to collect items in an inventory.
  *
- * <p>It keeps track of the currently carried items in the list {@link #inventory} and the size of
- * the inventory in {@link #maxSize}. Carried items can be retrieved by using {@link #items()
- * getItems}. List elements can be added via {@link #addItem(ItemData) addItem} and removed via
- * {@link #removeItem(ItemData) removeItem}.
+ * <p>The component stores a collection of {@link ItemData} that are currently stored in the
+ * associated entity.
  *
- * <p>The number of filled slots can be retrieved via {@link #filledSlots() filledSlots} and the
- * number of empty slots via {@link #emptySlots() emptySlots}. The maximum inventory size can also
- * be retrieved via {@link #maxSize() getMaxSize}.
+ * <p>Each inventory has a maximum number of items that can be carried.
+ *
+ * <p>Carried items can be retrieved using {@link #items() getItems}.
+ *
+ * <p>Items can be added via {@link #add(ItemData) addItem} and removed via {@link #remove(ItemData)
+ * removeItem}.
+ *
+ * <p>The number of items in the inventory can be retrieved using {@link #count() filledSlots}, and
+ * the number of items that can still be added can be retrieved using {@link #freeSpace()
+ * emptySlots}. The maximum inventory size can also be retrieved using {@link #maxSize()
+ * getMaxSize}.
  */
-public class InventoryComponent extends Component {
+public final class InventoryComponent extends Component {
 
-    private List<ItemData> inventory;
-    private int maxSize;
-    private final Logger inventoryLogger = Logger.getLogger(this.getClass().getName());
+    private final List<ItemData> inventory;
+    private final int maxSize;
+    private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
 
     /**
-     * Creates a new InventoryComponent
+     * Creates a new {@link InventoryComponent} with the given size and add it to the associated
+     * entity.
      *
-     * <p>Create a new InventoryComponent by explicitly setting the maximum inventory size.
-     *
-     * @param entity the Entity which will have the inventory added
-     * @param maxSize the maximal size of the inventory
+     * @param entity The associated entity.
+     * @param maxSize The number of items that can be stored in the inventory.
      */
-    public InventoryComponent(Entity entity, int maxSize) {
+    public InventoryComponent(final Entity entity, int maxSize) {
         super(entity);
         inventory = new ArrayList<>(maxSize);
         this.maxSize = maxSize;
     }
 
     /**
-     * Adding an Element to the Inventory does not allow adding more items than the size of the
-     * Inventory.
+     * Adds an item to the inventory.
```

Comment:
```suggestion
     * Add the given item to the inventory.
```

Reply:
Dumme Frage: Kann ich ein Item mehrfach in das Inventory legen, d.h. (a) sprechen wir hier von einer Liste oder einer Menge und (b) was passiert dann?

Reply:
Ist eine Liste und jetzt wo du es ansprichst ist das ein "bug".
Das muss ein Set sein. Ich kann zwar mehrere Heiltränke in mein Inventar ablegen, aber den selben Heiltrank nur einmal.

Reply:
Spannend! Daraus dann die nächste Frage: Wenn ich zwei verschiedene Heiltränke sammle, kann ich entweder die einzelnen Objekte speichern oder aber sagen "ich habe 2 Einträge mit dem Typ Heiltrank". Also belegen zwei Heiltrank-Objekte dann auch zwei Plätze in meinem Inventar oder ist das nur einer?

Reply:
Das hatten wir irgendwann mal definiert (kommt dann auch noch in die javadoc)
Jedes Objekt kostet einen Platz. Zwei Heiltränke daher zwei Plätze.
Hier können die Studis dann mit Taschen und Beutel etc. was besseres Entwickeln.

Reply:
sry, dass das schon wieder so viel arbeit wird.

Reply:
alles jut,
genau dafür sind diese PRs ja da. Um unklarheiten aufzuzeigen und zu klären/lösen/dokumentieren